shared/actions/fs/index.js

```diff
@@ -334,6 +334,58 @@ function* _loadMimeType(path: Types.Path) {
 
 const loadMimeType = (action: FsGen.MimeTypeLoadPayload) => Saga.call(_loadMimeType, action.payload.path)
 
+const commitEdit = (action: FsGen.CommitEditPayload, state: TypedState) => {
+  const {editID} = action.payload
+  const edit = state.fs.edits.get(editID)
+  if (!edit) {
+    return null
+  }
+  const {parentPath, name, type} = edit
+  switch (type) {
+    case 'new-folder':
```

Comment:
before i get too far into this pr, is this how you want this to be structured? a generic 'edit' with these sub types ? instead of it just being a high level type. aka why not just have this be a NewFolderPayload etc

Reply:
It seems to be something that can be shared between this and rename, and maybe upload (I.e. new entry) as well. Plus it needs to be plumbed into the store. I guess we could also have separate maps for new folder creation and renames if they were separate independent types, but then we still need to check for them separately in containers.

To be clear, I am planning to have separate types for rename, but both satisfying the Edit type.

Does that sound good to you?